### PR TITLE
[JENKINS-41434] Hide link when no create permission

### PR DESCRIPTION
**Pull request by imeredith**
# Description

To test create a user without admin and job create permissions.

See [JENKINS-41434](https://issues.jenkins-ci.org/browse/JENKINS-41434).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
This upsets a bunch of the ATH: 

https://ci.blueocean.io/job/ATH-Jenkinsfile/job/master/1575/#showFailuresLink

**Comment by michaelneale** [comment]
Hrm... i think this can't work with the ATH until it has user identity, this also stops blue ocean working for a whole bunch of people who used full anon access if I read it correctly. This may not be the time for this in retrospect.

**Comment by michaelneale** [comment]
ok have looked closer, yes this cases failures in the expected way. We can't do this one right now unfortunately without a bit of work to the ATH, or else putting "special casing" in for the ATH (which is not good). We don't know what else this may do yet either - the ATH runs with full anon access (and some people do) - and this will also affect them. 

So this is a :bug: for me from now, I think we need to postpone this (yes, we should have known this before starting). 

@imeredith tried to get you on various chat things but can't seem to reach you in the day - so can cover this later if you have any ideas.

**Comment by michaelneale** [comment]
@imeredith some other options: James suggests that the security check should allow that if they are admin (even anon but still admin) - but button could show. 

The other way is to alter the ATH just a little bit to not look for that button (short term thing).

**Comment by imeredith** [comment]
I think I have a 'fix' for ath which will let us login, will try it tomorrow.

**Comment by michaelneale** [comment]
https://github.com/jenkinsci/blueocean-acceptance-test/pull/122

should fix the test failure for this (very minor).

File: blueocean-core-js/src/js/User.js [hunk 1]
```diff
@@ -21,6 +98,18 @@ export class User {
...
+        const permission = this._blueUser._permission || {};
+        return !!permission.administrator;
+    }
+
+    get permissions() {
+        return {
+            pipeline: new PipelinePermissions(this._blueUser),
+            credential: new CredentialPermissions(this._blueUser),
+        };
+    }
```
Comment (cliffmeyers) [inline blueocean-core-js/src/js/User.js]: Do we want to add a getter or function for every single property in this graph? It seems like a maintenance problem as new permissions are added, we need to keep adding them here, and republishing every time. Should we consider doing a deep clone from the passed in `blueUser` directly onto the `User` instance?

**Comment by michaelneale** [comment]
@cliffmeyers good point - yes that wouldn't be a good experience to just let them fail at the end.

**Comment by michaelneale** [comment]
I think this should be postponed and ticket put back in queue until we have ATH - OR, as @i386 suggested, make this still show up in the case of security turned off (so it still is hidden when a logged in user doesn't have create permissions) - then ATH will be happy

**Comment by michaelneale** [comment]
This passes the ATH and is likely in better shape now - if updated then I think it can go in.

**Comment by cliffmeyers** [comment]
@michaelneale so we won't show an error message to the user if they've bookmarked creation and are not logged in? Could someone at least log that as a separate ticket that we can fix later?

**Comment by i386** [comment]
Could we just make it redirect?

**Comment by i386** [comment]
I raised https://issues.jenkins-ci.org/browse/JENKINS-42375 for this FYI

File: blueocean-core-js/src/js/User.js (hunk 1 above)
Comment (imeredith) [inline blueocean-core-js/src/js/User.js]: I agree I was just maintaining the api.

**Comment by michaelneale** [comment]
@imeredith this seems to fail when looking at the translation: https://ci.blueocean.io/job/ATH-Jenkinsfile/job/JENKINS-41434_creation_permissions/2/testReport/junit/(root)/loadGerman/Step_02/ - which is most odd... that shouldn't really be affected here - worth checking out though.

**Comment by michaelneale** [comment]
OK, after an "underarm bowl" incident where I made a branch on the ATH which ended up with the one failure tricking @imeredith  - this now passes ATH, and sniff test. 

:bee: from me.

**Pull request merged**